packages/aws-cdk/lib/api/hotswap/appsync-mapping-templates.ts

```diff
@@ -49,25 +50,76 @@ export async function isHotswappableAppSyncChange(
 
         const sdkProperties: { [name: string]: any } = {
           ...change.oldValue.Properties,
+          Definition: change.newValue.Properties?.Definition,
+          DefinitionS3Location: change.newValue.Properties?.DefinitionS3Location,
           requestMappingTemplate: change.newValue.Properties?.RequestMappingTemplate,
+          requestMappingTemplateS3Location: change.newValue.Properties?.RequestMappingTemplateS3Location,
           responseMappingTemplate: change.newValue.Properties?.ResponseMappingTemplate,
+          responseMappingTemplateS3Location: change.newValue.Properties?.ResponseMappingTemplateS3Location,
         };
         const evaluatedResourceProperties = await evaluateCfnTemplate.evaluateCfnExpression(sdkProperties);
         const sdkRequestObject = transformObjectKeys(evaluatedResourceProperties, lowerCaseFirstCharacter);
 
+        // resolve s3 location files as SDK doesn't take in s3 location but inline code
+        if (sdkRequestObject.requestMappingTemplateS3Location) {
+          sdkRequestObject.requestMappingTemplate = (await fetchFileFromS3(sdkRequestObject.requestMappingTemplateS3Location, sdk))?.toString('utf8');
```

Comment:
Because Synth produces assets with a hash in the filename, I think the S3Location *will* change if a resolver or function's code is updated, and yeah we use L1 constructs here.

@comcalvi not urgent, but we might actually want to open the discussion to using S3Location instead of the inline approach for the L2 (when an Asset is provided), since that means customers will far more quickly hit Cfn template size limits (we see this happen for Amplify customers w/o inlining much of these assets, since with AppSync it's easy to produce a large number of resources quickly, and if we're inlining all code then it's going to expand very quickly).

Reply:
ahh okay, so the s3 locations are managed through the CDK here. That makes sense, thanks for clarifying.